reader.go

```diff
@@ -1752,6 +1782,8 @@ func (r *reader) run(ctx context.Context, offset int64) {
 					r.withErrorLogger(func(log *log.Logger) {
 						log.Printf("the kafka reader is reading passed the last offset for partition %d of %s at offset %d", r.partition, r.topic, offset)
 					})
+					conn.Close()
```

Comment:
We should update the comment on the default case. As we are saying that something is wrong and we should just give up and re-init our connection

Reply:
Updated the comment.